boring/src/ssl/callbacks.rs

```diff
@@ -149,6 +178,33 @@ where
     callback(preverify_ok != 0, ctx) as c_int
 }
 
+pub(super) unsafe extern "C" fn ssl_raw_custom_verify<F>(
+    ssl: *mut ffi::SSL,
+    out_alert: *mut u8,
+) -> ffi::ssl_verify_result_t
+where
+    F: Fn(&mut SslRef) -> Result<(), SslVerifyError> + 'static + Sync + Send,
+{
+    // SAFETY: boring provides valid inputs.
+    let ssl = unsafe { SslRef::from_ptr_mut(ssl) };
+    let out_alert = unsafe { &mut *out_alert };
+
+    let callback = ssl
+        .ex_data(Ssl::cached_ex_index::<Arc<F>>())
```

Comment:
this is identical to `raw_custom_verify` except for how we get the ex_data index, can we extract the safe code into its own function?